src/com/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -480,20 +480,33 @@ private final void cast_cached_value(int new_type)
         final int count = _annotation_count;
         if (count == 0) return SymbolToken.EMPTY_ARRAY;
 
-        SymbolTable symbols = getSymbolTable();
+        resolveAnnotationSymbols();
 
         SymbolToken[] result = new SymbolToken[count];
-        for (int i = 0; i < count; i++)
-        {
-            SymbolToken sym = _annotations[i];
-            SymbolToken updated = _Private_Utils.localize(symbols, sym);
-            if (updated != sym) _annotations[i] = updated;
-            result[i] = updated;
-        }
+        System.arraycopy(_annotations, 0, result, 0, count);
 
         return result;
     }
 
+    public String[] getTypeAnnotations()
+    {
+        resolveAnnotationSymbols();
+        return _Private_Utils.toStrings(_annotations, _annotation_count);
+    }
+
+    /**
+     * Resolve annotations with the current symbol table.
+     */
+    private void resolveAnnotationSymbols() {
+        final int count = _annotation_count;
+        for (int i = 0; i < count; i++) {
+            SymbolTable symbols = getSymbolTable();
+            SymbolToken sym = _annotations[i];
+            SymbolToken updated = _Private_Utils.localize(symbols, sym);
+            if (updated != sym)
```

Comment:
please _always_ use { }, even for a single-line if block (and even if copied from existing code...)